src/c++/perf_analyzer/infer_data_manager_base.cc

```diff
@@ -30,6 +30,88 @@
 
 namespace triton { namespace perfanalyzer {
 
+cb::Error
+InferDataManagerBase::GetInputData(
+    const std::string& name, const ModelTensor& tensor, int stream_id,
+    int step_id, std::vector<const uint8_t*>& data_ptrs,
+    std::vector<size_t>& byte_size)
+{
+  size_t max_count = tensor.is_shape_tensor_ ? 1 : batch_size_;
+  std::vector<int64_t> shape;
+  std::vector<int64_t> prev_shape;
+
+  for (size_t count = 0; count < max_count; count++) {
+    int local_step_id =
+        (step_id + count) % data_loader_->GetTotalSteps(stream_id);
+    const uint8_t* data_ptr{nullptr};
+    size_t batch1_bytesize;
+
+    RETURN_IF_ERROR(
+        data_loader_->GetInputShape(tensor, stream_id, local_step_id, &shape));
+    if (!shape.empty()) {
+      if (count == 0) {
+        prev_shape = shape;
+      } else {
+        if (!std::equal(shape.begin(), shape.end(), prev_shape.begin())) {
+          return cb::Error(
+              "can not batch tensors with different shapes together "
+              "(input '" +
+                  name + "' expected shape " + ShapeVecToString(prev_shape) +
+                  " and received " + ShapeVecToString(shape),
+              pa::GENERIC_ERROR);
+        }
+      }
+    }
+
+    RETURN_IF_ERROR(data_loader_->GetInputData(
+        tensor, stream_id, local_step_id, &data_ptr, &batch1_bytesize));
+
+    data_ptrs.push_back(data_ptr);
+    byte_size.push_back(batch1_bytesize);
+  }
+
+  return cb::Error::Success;
+}
+
+cb::Error
+InferDataManagerBase::ValidateShapeTensor(
+    const ModelTensor& tensor, int stream_id, int step_id,
+    const std::vector<const uint8_t*>& data_ptrs,
+    const std::vector<size_t>& byte_size)
+{
+  // Validate if the shape tensors specified in the batch are identical.
+  for (size_t count = 1; count < batch_size_; count++) {
```

Comment:
why starting at 1 here?
id 0 is intrinsically valid?

Reply:
I updated the comment for clarity